### PR TITLE
Security: stop panicking on out-of-range version timestamps, Credit: Equilibrium

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

If Zebra receives an out-of-range time in a version message, it will panic.

This issue was reported by Niklas Long of Equilibrium.

## Solution

Instead, return a deserialization error, and close the connection.

The code in this pull request has:
  - [x] Documentation Comments
  - [x] Unit Tests

## Root Causes

`chrono`'s `timestamp` function panics, and the `timestamp_opt` function returns a `Result`-like enum. But most other Rust APIs have an explicit `*_unwrap` suffix on functions that panic.

Bitcoin's network protocol uses `u32` for most times, but `i64` for times in the version message. All `u32`s are valid `chrono` times, but some `i64`s are invalid.

## Review

@conradoplg would you like to review this fix? It's not particularly urgent, because it doesn't affect other nodes.

@niklaslong how would you like to be credited in the Zebra release notes?

## Related Issues

I'm just about to post a cleanup PR for Zebra's other time serialization.

### CODE REVIEW

File: zebra-network/src/protocol/external/codec.rs [hunk 1]
```diff
@@ -678,7 +695,73 @@ mod tests {
...
+            {
+                let mut fw = FramedWrite::new(&mut bytes, Codec::builder().finish());
+                fw.send(v.clone())
+                    .await
+                    .expect("message should be serialized");
+            }
+
+            let old_bytes = bytes.clone();
+
+            // tweak the version bytes so they're out of range
```
Comment (teor2345) [inline zebra-network/src/protocol/external/codec.rs]: Oops
```suggestion
            // tweak the version bytes so the timestamp is set to `time`
```

**Review by conradoplg — APPROVED** [review]
Looks good.

Should we wait for @niklaslong to confirm how we'll credit before merging?

**Comment by teor2345** [comment]
> Looks good.
> 
> Should we wait for @niklaslong to confirm how we'll credit before merging?

Yeah I'll ping them now.

**Comment by teor2345** [comment]
I'm going to put this PR in draft until we've confirmed the security disclosure credit.

**Comment by niklaslong** [comment]
> Should we wait for @niklaslong to confirm how we'll credit before merging?

Thanks for asking! My preference would be for the credit to go to Equilibrium.

**Pull request merged**